apps/voting-disputable/contracts/DisputableVoting.sol

```diff
@@ -0,0 +1,745 @@
+        votingPower = vote_.votingPower;
+        script = vote_.executionScript;
+    }
+
+    /**
+    * @dev Return the disputable information for a vote by its ID
+    * @param _voteId Vote identifier
+    * @return Vote agreement action ID
+    * @return Vote paused date
+    * @return Vote paused duration
+    * @return Vote disputable status
+    */
+    function getDisputableInfo(uint256 _voteId)
+        external
+        view
+        returns (
+            uint256 actionId,
+            uint64 pausedAt,
+            uint64 pauseDuration,
+            DisputableStatus status
+        )
+    {
+        Vote storage vote_ = votes[_voteId];
+        actionId = vote_.actionId;
+        pausedAt = vote_.pausedAt;
+        pauseDuration = vote_.pauseDuration;
+        status = vote_.disputableStatus;
+    }
+
+    /**
+    * @notice Tells whether a vote #`_voteId` is open or not
+    * @dev Initialization check is implicitly provided by `voteExists()` as new votes can only be
+    *      created via `newVote()`, which requires initialization
+    * @param _voteId Id for vote
+    * @return True if the vote is still open, false otherwise
+    */
+    function isVoteOpen(uint256 _voteId) public view voteExists(_voteId) returns (bool) {
+        return _isVoteOpen(votes[_voteId]);
+    }
+
+    /**
+    * @dev Return the state of a voter for a given vote by its ID
+    * @param _voteId Vote identifier
+    * @param _voter Address of the voter
+    * @return VoterState of the requested voter for a certain vote
+    */
+    function getVoterState(uint256 _voteId, address _voter) public view voteExists(_voteId) returns (VoterState) {
+        return _voterState(votes[_voteId], _voter);
+    }
+
+    /**
+    * @dev Return the caster of a given vote by its ID
+    * @param _voteId Vote identifier
+    * @param _voter Address of the voter
+    * @return Address of the caster of the voter's vote
+    */
+    function getVoteCaster(uint256 _voteId, address _voter) public view voteExists(_voteId) returns (address) {
+        return _voteCaster(votes[_voteId], _voter);
+    }
+
+    /**
+    * @notice Tells whether `_representative` is `_voter`'s representative or not
+    * @return True if the given representative was allowed by a certain voter, false otherwise
+    */
+    function isRepresentativeOf(address _voter, address _representative) public view isInitialized returns (bool) {
+        return _isRepresentativeOf(_voter, _representative);
+    }
+
+    /**
+    * @notice Tells whether vote #`_voteId` is within the overrule period for delegated votes or not
+    * @dev Initialization check is implicitly provided by `voteExists()` as new votes can only be
+    *      created via `newVote(),` which requires initialization
+    * @return True if the requested vote is within the overrule window for delegated votes, false otherwise
+    */
+    function withinOverruleWindow(uint256 _voteId) public view voteExists(_voteId) returns (bool) {
```

Comment:
I'm wondering if we should just make all the getters `external`; I don't think we'll see much inheritance in practice with the size limit, and there are always the internal methods exposed anyway.

On that note, maybe we should make all the internal ones `private` too, to close this off entirely from extension?

Reply:
I agree about the `external` ones, but I wouldn't commit to the `internal`s. Why would you avoid people extending this app?

Reply:
Two reasons:

1. Extensions are naturally difficult to create and audit, as you have less control over the upstream contract (you're probably better off just copying and updating the original contract)
2. We're most likely not leaving a lot of room for extension anyway, so we should do what we can to make this one cheaper / lighter

But yes, I think between `internal` and `private` solidity doesn't generate different code, so it's probably fine to leave as `internal` (but then the current `private` methods should probably just be made `internal`).